File: 0459-repeated-substring-pattern/0459-repeated-substring-pattern.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        for i in range(len(s) // 2, 0, -1):
            if len(s) % i != 0:
                continue
            if s[:i:] * (len(s) // i) == s:
                return True
        return False

File: 0459-repeated-substring-pattern/0459-repeated-substring-pattern.cpp
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        string s_ = s;
        char c;
        for(int i = 0; i < s.length() - 1; ++i) {
            c = s_[s_.length()-1];
            s_.pop_back();
            s_ = c + s_;
            if(s == s_) {
                return true;
            }
        }
        return false;
    }
};